### PR TITLE
Implement Sub-MOTD Feature

### PR DESCRIPTION
**Pull request by xRodzMC**
# What is this?
This is plugin is target to test the Sub-Motd Support for PocketMine-MP.<br>
This is on-going to be test in the future, if this repository was archived, that means it was approved to the PocketMine-MP Original Repository.
# What is Sub-MOTD?
Custom Sub-MOTD Support as on Custom MOTD in the config. This changes the interface query of the server.
[](https://github.com/pmmp/PocketMine-MP/issues/5204)
The "My World" will change as on `server.properties` provided sub-motd. We can say its useless but if we add this feature we can say people will not going to make their own pocketmine-mp for changing `VersionInfo.php` the name. So we can use function `Network->setName();` & `Network->setLanName()` if value is missing it will returning `VersionInfo::NAME` as default. People wont make their own useless Pocketmine-MP Forks to implement this feature.
- https://github.com/pmmp/PocketMine-MP/pull/4737
- [Sub-motd (description)? (PMMP Forums)](https://forums.pmmp.io/threads/sub-motd-description.10185/)

# Changes
## API
```php
Network->setName("MOTD");
Network->setLanName("MOTD");
```
## Behaivor Changes
[](https://github.com/pmmp/PocketMine-MP/issues/5204)

## Backwards Compability
```php
$this->getServer()->getNetwork()->setName("MOTD");
$this->getServer()->getNetwork()->setLanName("MOTD");

$this->getServer()->getMotd();
$this->getServer()->getLanMotd();

$this->getServer()->getNetwork()->getName();
$this->getServer()->getNetwork()->getLanName();

$this->getServer()->getQueryInformation()->getLanServerName();
$this->getServer()->getQueryInformation()->setLanServerName("MOTD");
```

# Tests
Test the server and run with example plugin test for this feature.
- [ExamplePlugin](https://github.com/xqwtxon/SubMotd-TestPlugin)

# For Plugin Developers
Do not use `Network->internalSetName()`, this kindly mostly bug on PocketMine-MP RakLib, The Query is displaying the ipv4/ipv6. It break backwards compability and behaivor of the game. Take a example how developers can do with it.

# For Internal Developers
Please help me to fix this `internalSetName()` to protect these agains the BC Break.

### CODE REVIEW

**Comment by xRodzMC** [comment]
Sorry, if my commit is unverified, i havent setup yet my gpg key.

File: src/PocketMine.php [hunk 1]
```diff
@@ -245,7 +245,7 @@ function server(){
 				critical_error("Please synchronize Composer dependencies before running the server.");
 				exit(1);
 			}
-		}
+		}*/
```
Comment (NhanAZ) [inline src/PocketMine.php]: Why did you put this in the comment?

File: tests/phpunit/block/BrewingStandTest.php [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 use PHPUnit\Framework\TestCase;
 use pocketmine\block\utils\BrewingStandSlot;
+use function array_values;
```
Comment (NhanAZ) [inline tests/phpunit/block/BrewingStandTest.php]: This has nothing to do with PR

File: src/PocketMine.php (hunk 1 above)
Comment (xRodzMC) [reply src/PocketMine.php]: oh my bad

File: tests/phpunit/block/BrewingStandTest.php (hunk 2 above)
Comment (xRodzMC) [reply tests/phpunit/block/BrewingStandTest.php]: I hate php-cs-fixer dryrun ๏_๏

**Comment by Endermanbugzjfc** [comment]
What is a “sub-MOTD”?

**Comment by NhanAZ** [comment]
> What is a “sub-MOTD”?

https://github.com/pmmp/PocketMine-MP/issues/5204

**Comment by Endermanbugzjfc** [comment]
> > What is a “sub-MOTD”?
> 
> 
> 
> https://github.com/pmmp/PocketMine-MP/issues/5204

Thanks.

**Review by Gaprix — APPROVED** [review]

**Review by dktapps — CHANGES_REQUESTED** [review]
As before, the documentation available for this feature is inadequate. Most of the time, people won't even see this text, so they'll think something is broken when it doesn't appear to change anything.

**Comment by matcracker** [comment]
Since this MOTD is only shown in the "LAN" page list, wouldn't it be better to call the parameter as "lan-motd"? At first not knowing what the sub-motd was, I was expecting a message under the current MOTD instead it only concerns the LAN screen.

**Comment by xRodzMC** [comment]
> Since this MOTD is only shown in the "LAN" page list, wouldn't it be better to call the parameter as "lan-motd"? At first not knowing what the sub-motd was, I was expecting a message under the current MOTD instead it only concerns the LAN screen.

Good idea :)

**Comment by xRodzMC** [comment]
For Developers:

Please do not use `Network->internalSetName()` It breaks backwards compabilities. If you want to improve this or change this, please kindly make it at your own. If you want to help us, please kindly create your own pull requests.

**Comment by xRodzMC** [comment]
Also, i found more links about the pull requests, such as forum and other pull requests.

File: src/PocketMine.php (hunk 1 above)
Comment (xRodzMC) [reply src/PocketMine.php]: fixed

**Comment by xRodzMC** [comment]
My commits are so advance time, sorry but time havent configured yet.

**Comment by dktapps** [comment]
This PR has taken up way too much time and effort to be worthwhile for such a small change. It's also now venturing into BC-break territory since it adds new requirements to the NetworkInterface interface. In addition, the author seems unwilling or unable to address review comments, so I'm going to close this.

**Comment by xRodzMC** [comment]
> This PR has taken up way too much time and effort to be worthwhile for such a small change. It's also now venturing into BC-break territory since it adds new requirements to the NetworkInterface interface. In addition, the author seems unwilling or unable to address review comments, so I'm going to close this.

Sorry, i have enough time to do work with it, Since i was busy...

**Pull request closed without merging**